### ISSUE TITLE
closing brackets give up the current goal

### ISSUE THREAD

**Issue by Zimmi48** [Zimmi48/bugzilla-test#5642]
Note: the issue was created automatically with bugzilla2github tool

# Bugzilla Bug ID: 5642
Date: 2017-07-12 15:59:20 +0200
From: julien.lepiller@inria.fr
To:   
CC:   coq-bugs-redist@lists.gforge.inria.fr, @_silene

Last updated: 2017-07-28 14:50:08 +0200

## Bugzilla Comment ID: 17048
Date: 2017-07-12 15:59:20 +0200
From: julien.lepiller@inria.fr

When using coqIDE or coquille, a closing bracket will automatically give up the current goal, while in proof-general or coqc you get an error.

for instance, the following code will be accepted by coqIDE and coquille, but not by proof-general and will not compile with coqc:

Lemma a: 0 < 1.
Proof.
  { idtac. }
Admitted.

Although the following gives the expected error in coqIDE:

Lemma b: 0 < 1.
Proof.
  { }
Admitted.

After looking a bit, I found that using coqtop interactively with no argument will produce the expected error in both cases, while using coqtop with the "-ideslave -main-channel stdfds -async-proofs on" arguments (that's what coquille uses) will produce an error only in the second case.

## Bugzilla Comment ID: 17190
Date: 2017-07-28 14:50:08 +0200
From: @_silene

As strange as it may seem, this is actually a feature. See the documentation of the -async-proofs-tactic-error-resilience option in the reference manual: https://coq.inria.fr/distrib/current/refman/Reference-Manual031.html

Basically, Coq will heuristically add "admit" to a proof so that the user can keep working on a proof script even if an error has already been encountered.
